common/test/acceptance/tests/lms/test_lms_dashboard.py

```diff
@@ -29,26 +29,63 @@ def setUp(self):
         self.dashboard_page = DashboardPage(self.browser)
 
         # Configure some aspects of the test course and install the settings into the course
-        self.course_fixture = CourseFixture(
-            self.course_info["org"],
-            self.course_info["number"],
-            self.course_info["run"],
-            self.course_info["display_name"],
-        )
-        self.course_fixture.add_advanced_settings({
-            u"social_sharing_url": {u"value": "http://custom/course/url"}
-        })
-        self.course_fixture.install()
-
-        # Create the test user, register them for the course, and authenticate
+        self.courses = {
+            'A': {
+                'org': 'test_org',
+                'number': self.unique_id,
+                'run': 'test_run_A',
+                'display_name': 'Test Course A'
+            },
+            'B': {
+                'org': 'test_org',
+                'number': self.unique_id,
+                'run': 'test_run_B',
+                'display_name': 'Test Course B'
+            },
+            'C': {
+                'org': 'test_org',
+                'number': self.unique_id,
+                'run': 'test_run_C',
+                'display_name': 'Test Course C'
+            }
+        }
+
         self.username = "test_{uuid}".format(uuid=self.unique_id[0:6])
         self.email = "{user}@example.com".format(user=self.username)
-        AutoAuthPage(
-            self.browser,
-            username=self.username,
-            email=self.email,
-            course_id=self.course_id
-        ).visit()
+
+        self.course_keys = {}
+        self.course_fixtures = {}
+
+        for key, value in self.courses.iteritems():
+            self.course_key = generate_course_key(
+                value['org'],
+                value['number'],
+                value['run'],
+            )
+
+            self.course_fixture = CourseFixture(
```

Comment:
Again, in this loop you are repeatedly assigning to self.XXX, which doesn't makes sense. Whatever happens to be last in the loop will remain as the value for the test cases, and it is very unexpected.